src/peft/tuners/tuners_utils.py

```diff
@@ -361,7 +361,15 @@ def _check_merge_allowed(self):
 
         Raise a ValueError if it is not possible to merge the adapter with the given configuration.
         """
-        pass
+        tied_target_modules = self._get_tied_target_modules(self.model)
+        if tied_target_modules:
+            warnings.warn(
+                f"Model with `tie_word_embeddings=True` and the {tied_target_modules=} are part of the adapter. "
+                "This can lead to complications when merging the adapter. "
+                "You can opt to merge the adapter after cloning the weights (to untie the embeddings), "
```

Comment:
Honestly, I didn't know about the option to pass `tie_word_embeddings=False`. Is there even a need to clone the weights in that case?